script.tablo/lib/util.py

```diff
@@ -0,0 +1,508 @@
+    return '0s'
+
+
+SIZE_NAMES = ("B", "KB", "MB", "GB", "TB", "PB", "EB", "ZB", "YB")
+
+
+def simpleSize(size):
+    """
+    Converts bytes to a short user friendly string
+    Example: 12345 -> 12.06 KB
+    """
+    s = 0
+    if size > 0:
+        i = int(math.floor(math.log(size, 1024)))
+        p = math.pow(1024, i)
+        s = round(size / p, 2)
+    if (s > 0):
+        return '%s %s' % (s, SIZE_NAMES[i])
+    else:
+        return '0B'
+
+
+class LoadingDialogWindow(xbmcgui.WindowXMLDialog):
+    def __init__(self, *args, **kwargs):
+        self.abort = False
+        self._winID = None
+
+    def onInit(self):
+        self._winID = xbmcgui.getCurrentWindowDialogId()
+
+    def onAction(self, action):
+        try:
+            if action == xbmcgui.ACTION_NAV_BACK or action == xbmcgui.ACTION_PREVIOUS_MENU:
+                self.abort = True
+                self.setProperty('abort', '1')
+                return
+        except:
+            ERROR()
+
+        xbmcgui.WindowXMLDialog.onAction(self, action)
+
+    def setProperty(self, key, value):
+        try:
+            xbmcgui.WindowXMLDialog.setProperty(self, key, value)
+            xbmcgui.Window(self._winID).setProperty(key, value)
+        except RuntimeError:
+            import traceback
+            traceback.print_exc()
+
+
+class LoadingDialog(object):
+    def __init__(self):
+        self.w = None
+        self._event = threading.Event()
+
+    def __enter__(self):
+        return self.show()
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        self.close()
+
+    def show(self):
+        self.w = LoadingDialogWindow('script-tablo-loading.xml', ADDON.getAddonInfo('path'), 'Main')
+        self.w.show()
+        self._event.clear()
+        return self
+
+    def close(self):
+        self._event.set()
+
+        if self.w:
+            self.w.close()
+            del self.w
+
+        self.w = None
+
+    @property
+    def canceled(self):
+        if self.w:
+            return self.w.abort
+        else:
+            return False
+
+    def wait(self):
+        self._event.wait()
+
+
+def busyDialog(func):
+    def inner(*args, **kwargs):
+        w = None
+        try:
+            w = xbmcgui.WindowXMLDialog('script-tablo-busy.xml', ADDON.getAddonInfo('path'), 'Main')
+            w.show()
+            return func(*args, **kwargs)
+        finally:
+            if w:
+                w.close()
+            del w
+    return inner
+
+
+def withBusyDialog(method, *args, **kwargs):
+    return busyDialog(method)(*args, **kwargs)
+
+
+def loadingDialog(func):
+    def inner(*args, **kwargs):
+        w = None
+        try:
+            w = xbmcgui.WindowXMLDialog('script-tablo-loading.xml', ADDON.getAddonInfo('path'), 'Main')
+            w.show()
+            return func(*args, **kwargs)
+        finally:
+            if w:
+                w.close()
+            del w
+    return inner
+
+
+def withLaodingDialog(method, *args, **kwargs):
```

Comment:
Turns out it's not even used :) I probably was using and dropped it. I'll remove this for the _next_ version if it's all the same.